### DISCUSSION TITLE
usb endpoint enumeration linux kernel dirver fix

### DISCUSSION THREAD

**Discussion by KozhinovAlexander** [CANnectivity/cannectivity#25] [Ideas]
Meanwhile the implementation of linux kernel patch seems to be shaped well (see the comments from Vincent Mailhol [here](https://lore.kernel.org/all/CAMZ6Rq+AfFvrRG4muTyK6xrOQ-Wa8n8Ha5XG=yesna+1FGHXzw@mail.gmail.com/)). 
Also the v4 (hopefully last one) can be found [here](https://lore.kernel.org/all/20241012112522.18464-1-ak.alexander.kozhinov@gmail.com/).

@henrikbrixandersen With the change from above patch [usb dummy ep](https://github.com/CANnectivity/cannectivity/blob/main/subsys/usb/device/class/gs_usb.c#L1480) could be dropped now. The linux-kernel patch is also backward compatible and works with candeLight_fw I use on my MKS Canable V2 board. 
I would like to add this changes to this repository. To address backward compatibility I think to add a KConfig switch s.t. dummy ep can be disabled. What do you think about it?

@henrikbrixandersen FYI: Somehow I was not able to open discussuon in this project - thus I've opened an issue. Please feel free move it to discussion.

**Comment by henrikbrixandersen** [comment]
> I would like to add this changes to this repository. To address backard compatibility I think to add KConfig switch s.t. dummy ep can be disabled.

I am already working on a similar addition, but thanks.

**Reply by KozhinovAlexander** [reply]
Do you plan to add changes in Linux Kernel too or only in the CANnectivity project?

**Reply by henrikbrixandersen** [reply]
I am only planning to change the CANnectivity part. Your proposed patch to the Linux kernel driver looks fine.

**Reply by KozhinovAlexander** [reply]
> I am only planning to change the CANnectivity part. Your proposed patch to the Linux kernel driver looks fine. 

It sounds great. Thank you. Please mention me if you do a PR. Happy to see it soon.

**Comment by KozhinovAlexander** [comment]
Resolved by: https://github.com/CANnectivity/cannectivity/discussions/25#discussioncomment-10922920
